compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -261,6 +266,221 @@ static CodeGenPipeline getTensorCorePipeline(Type elementType) {
+/// dispatch. The reduction op shouldn't have any consumer
+/// because that may introduce new dimensions and the
+/// vector distribution analysis might fail. The lowering config is
+/// only attached to the reduction op for now.
+/// TODO(pashu123): Analyze the shared parallel and reduction
+/// dimensions within a dispatch. Based on the analysis try to
+/// put the lowering_config on each of the operation within a
+/// dispatch. That way we can support multiple reduction ops
+/// within a dispatch.
+static LogicalResult
+setReductionVectorDistributionConfig(IREE::GPU::TargetAttr target,
+                                     mlir::FunctionOpInterface entryPoint,
+                                     linalg::LinalgOp op) {
+
+  int64_t countReduction = 0;
+
+  // TODO(pashu123): Remove this check and allow multiple reductions in a single
+  // dispatch.
+  WalkResult walkResult = entryPoint.walk([&](Operation *op) {
+    if (auto linalgOp = dyn_cast<linalg::LinalgOp>(op)) {
+      if (hasReductionIterator(linalgOp)) {
+        countReduction++;
+      }
+    }
+    if (countReduction > 1) {
+      return WalkResult::interrupt();
+    }
+    return WalkResult::advance();
+  });
+
+  if (walkResult.wasInterrupted()) {
+    return failure();
+  }
+
+  // TODO(pashu123): The consumer operation may introduce new
+  //  dimension which will fail the distribution analysis.
+  for (auto potentialLinalgUser : op->getUsers()) {
+    if (auto linalgOp = dyn_cast<linalg::LinalgOp>(potentialLinalgUser)) {
+      return failure();
+    }
+  }
+
+  SmallVector<unsigned> parallelDims;
+  SmallVector<unsigned> reductionDims;
+  op.getParallelDims(parallelDims);
+  op.getReductionDims(reductionDims);
+
+  if (reductionDims.empty()) {
+    return failure();
+  }
+
+  SmallVector<int64_t, 4> bounds = op.getStaticLoopRanges();
+  IREE::GPU::TargetWgpAttr wgp = target.getWgp();
+
+  bool foundSingleReductionOutput = false;
+  for (auto [index, initOpOperand] : llvm::enumerate(op.getDpsInitsMutable())) {
+    // Only single combiner operations are supported for now.
+    SmallVector<Operation *> combinerOps;
+    if (matchReduction(op.getRegionOutputArgs(), index, combinerOps) &&
+        combinerOps.size() == 1) {
+      if (foundSingleReductionOutput)
+        return failure();
+      foundSingleReductionOutput = true;
+      continue;
+    }
+    if (!op.getMatchingIndexingMap(&initOpOperand).isIdentity())
+      return failure();
+  }
+  if (!foundSingleReductionOutput)
+    return failure();
+
+  SmallVector<int64_t> workgroupTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> threadTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> partialReductionTileSizes(op.getNumLoops(), 0);
+  SmallVector<int64_t> threadCounts(op.getNumLoops(), 1);
+  SmallVector<int64_t> subgroupCounts(op.getNumLoops(), 1);
+  SmallVector<int64_t> mapping(op.getNumLoops());
+  std::iota(mapping.begin(), mapping.end(), 0);
```

Comment:
nit: this could also be `llvm::to_vector<int64_t>(seq(0, op.getNumLoop()))`, but fine to leave as-is